packages/amplify-category-auth/src/events/prePushHandler.ts

```diff
@@ -0,0 +1,21 @@
+import { $TSContext, AmplifyError } from '@aws-amplify/amplify-cli-core';
+import { AuthInputState } from '../provider-utils/awscloudformation/auth-inputs-manager/auth-input-state';
+import { getAuthResourceName } from '../utils/getAuthResourceName';
+
+export const prePushHandler = async (context: $TSContext): Promise<void> => {
+  let authResourceName: string;
+  try {
+    authResourceName = await getAuthResourceName(context);
+  } catch {
+    // if the project doesn't have auth, early return
+    return;
+  }
```

Comment:
this might catch legit errors.
should we do something like https://github.com/aws-amplify/amplify-cli/blob/78590b8d4face7bbc0b174337b77b118affc525b/packages/amplify-category-auth/src/provider-utils/awscloudformation/utils/project-has-auth.ts#L7-L14 instead ?

Reply:
That's basically what getAuthResourceName is doing. It's using `context.amplify.getResourceStatus` instead of `stateManager.getMeta()` but if that throws an error the push is going to fail anyway because that same method is used to determine what changes there are to push.

Reply:
Although I could just use `projectHasAuth`